File: frontend/views/ueberfuehredenboesewicht/ueberfuehreden-boesewicht-view.ts
import { html, LitElement, customElement } from 'lit-element';



@customElement('ueberfuehreden-boesewicht-view')
export class UeberfuehredenBoesewichtView extends LitElement {
  createRenderRoot() {
    // Do not use a shadow root
    return this;
  }
  render() {
    return html`
        <vaadin-horizontal-layout class="content" style="width: 100%; align-items: flex-start; justify-content: center; height: 100%;">
            <vaadin-vertical-layout theme="spacing" style="align-self: flex-start; align-items: center; justify-content: flex-start; width: 5%;"></vaadin-vertical-layout>
            <vaadin-vertical-layout style="align-self: center; align-items: center; justify-content: center; width: 70%; height: 100%; padding: var(--lumo-space-m);">
                <h2 id="header">Überführe den Bösewicht</h2>
                <h3 id="textarea">Jetzt wird es ernst! Wir haben ein Bild des Bösewichts schießen können, auf dem seine Deepfake Software nicht richtig funktioniert hat. Dadurch sind einzelne Merkmale des Bösewichts erkennbar. Tipp: Achte auf Aspekte wie die Haar- oder Augenfarbe.</h3>
              <img id="corruptedImg" style="width: 50%; align-self: center;" src="target/classes/com/pip/deepfakes/views/res/images/ueberfuehre_den_boesewicht/corrupted.png">
                <vaadin-horizontal-layout theme="spacing" style="flex-grow: 1; flex-shrink: 1; width: 100%; justify-content: center; height: 10%;">
                  <vaadin-button id="readyButton"></vaadin-button>
                  <img id="taeter1" class="click" style="width: 20%; align-self: flex-start;" src="target/classes/com/pip/deepfakes/views/res/images/ueberfuehre_den_boesewicht/verdachtiger_1.jpg">
                  <img id="taeter2" class="click" style="width: 20%; align-self: flex-start;" src="target/classes/com/pip/deepfakes/views/res/images/ueberfuehre_den_boesewicht/verdachtiger_2.jpg">
                  <img id="taeter3" class="click" style="width: 20%; align-self: flex-start;" src="target/classes/com/pip/deepfakes/views/res/images/ueberfuehre_den_boesewicht/verdachtiger_3.jpg">
                  <img id="taeter4" class="click" style="width: 20%; align-self: flex-start;" src="target/classes/com/pip/deepfakes/views/res/images/ueberfuehre_den_boesewicht/verdachtiger_4.jpg">
                </vaadin-horizontal-layout>
            </vaadin-vertical-layout>
            <vaadin-vertical-layout theme="spacing" style="align-self: flex-start; align-items: center; justify-content: flex-start; width: 15%;"></vaadin-vertical-layout>
        </vaadin-horizontal-layout>
    `;
  }
}

File: src/main/java/com/pip/deepfakes/views/ueberfuehredenboesewicht/UeberfuehredenBoesewichtView.java
package com.pip.deepfakes.views.ueberfuehredenboesewicht;

import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.html.H2;
import com.vaadin.flow.component.html.H3;
import com.vaadin.flow.component.html.Image;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.polymertemplate.Id;
import com.vaadin.flow.router.Route;
import com.vaadin.flow.router.PageTitle;
import com.pip.deepfakes.views.MainLayout;
import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;

@PageTitle("Überführe den Bösewicht")
@Route(value = "solve", layout = MainLayout.class)
@Tag("ueberfuehreden-boesewicht-view")
@JsModule("./views/ueberfuehredenboesewicht/ueberfuehreden-boesewicht-view.ts")
public class UeberfuehredenBoesewichtView extends LitTemplate {

    // This is the Java companion file of a design
    // You can find the design file inside /frontend/views/

    @Id("taeter1")
    private Image tater1;

    @Id("taeter2")
    private Image tater2;

    @Id("taeter3")
    private Image tater3;

    @Id("taeter4")
    private Image tater4;

    @Id("readyButton")
    private Button readyButton;

    @Id("textarea")
    private H3 textArea;

    @Id("header")
    private H2 header;

    @Id("corruptedImg")
    private Image corrupted;


    public UeberfuehredenBoesewichtView() {
        reset();
    }

    private void setUpDecision() {
        readyButton.setVisible(false);
        textArea.setVisible(false);
        corrupted.setVisible(true);
        tater1.setVisible(true);
        tater2.setVisible(true);
        tater3.setVisible(true);
        tater4.setVisible(true);
        tater2.addClickListener(event -> {
//            MainLayout.tabs.setSelectedTab(MainLayout.tabs_instances.get(4));
            tater2.getUI().ifPresent(ui -> ui.navigate("thankyou"));
//            MainLayout.makeProgress(5);
        });
        tater1.addClickListener(imageClickEvent -> handleWrongDecision());
        tater3.addClickListener(imageClickEvent -> handleWrongDecision());
        tater4.addClickListener(imageClickEvent -> handleWrongDecision());
    }

    private void handleWrongDecision() {
        corrupted.setVisible(false);
        tater1.setVisible(false);
        tater2.setVisible(false);
        tater3.setVisible(false);
        tater4.setVisible(false);
        textArea.setVisible(true);
        readyButton.setVisible(true);
        header.setText("Schade");
        textArea.setText("Das war leider nicht der Täter. Versuche es nochmal! Tipp: Achte auf Aspekte wie die Haar- T-shirt- oder Augenfarbe.");
        readyButton.setText("Nochmal versuchen");
        readyButton.addClickListener(event -> {
            readyButton.getUI().ifPresent(ui -> reset() );
        });
    }

    private void reset() {
        header.setText("Überführe den Bösewicht");
        textArea.setVisible(true);
        textArea.setText("Jetzt wird es ernst! Wir haben ein Bild des Bösewichts schießen können, auf dem seine Deepfake Software nicht richtig funktioniert hat. Dadurch sind einzelne Merkmale des Bösewichts erkennbar.");
        readyButton.setVisible(true);
        readyButton.setText("Ich bin bereit");
        readyButton.addClickListener(event -> setUpDecision());
        corrupted.setVisible(false);
        tater1.setVisible(false);
        tater2.setVisible(false);
        tater3.setVisible(false);
        tater4.setVisible(false);
    }
}
